solidity/contracts/BancorNetwork.sol

```diff
@@ -640,10 +691,10 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractIds, FeatureIds {
 
         // if the allowance is nonzero, must reset it to 0 first
         if (_token.allowance(this, _spender) != 0)
-            assert(_token.approve(_spender, 0));
+            INonStandardERC20(_token).approve(_spender, 0);
```

Comment:
so worst case the tx will fail?

Reply:
when calling transferFrom afterwards, if the approve didn't succeed the tx should revert